File: volley/src/main/java/com/android/volley/toolbox/EncryptionHelper.java
package com.android.volley.toolbox;


import android.text.TextUtils;
import android.util.Base64;
import android.util.Log;

import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;

/**
 * Created by ADMIN on 7/19/2016.
 */
public class EncryptionHelper {

    private static final String TAG = "EncryptionHelper";
    private static final String ALGO = "AES";

    public static String encrypt(byte[] plainText, String key) {
        if (!TextUtils.isEmpty(key)) {
            try {
                byte[] authKey = key.getBytes();
                Cipher cipher = Cipher.getInstance(ALGO);
                SecretKeySpec secretKey = new SecretKeySpec(authKey, ALGO);
                cipher.init(Cipher.ENCRYPT_MODE, secretKey);
                byte[] cipherText = cipher.doFinal(plainText);
                String encryptedString = new String(Base64.encode(cipherText, Base64.DEFAULT), "UTF-8");
                return encryptedString.replaceAll("\n", "");
            } catch (Exception e) {
                Log.e(TAG, "encrypt: ", e);
            }
        }
        return null;
    }

    public static String encrypt(String plainText, String key) {
        if (!TextUtils.isEmpty(plainText) && !TextUtils.isEmpty(key))
            return encrypt(plainText.getBytes(), key);
        else
            return null;
    }

    public static String decrypt(String encryptedText, String key) {
        if (!TextUtils.isEmpty(key)) {
            try {
                byte[] authKey = key.getBytes();
                Cipher cipher = Cipher.getInstance(ALGO);
                SecretKeySpec secretKey = new SecretKeySpec(authKey, ALGO);
                cipher.init(Cipher.DECRYPT_MODE, secretKey);
                byte[] cipherText = Base64.decode(encryptedText.getBytes("UTF8"), Base64.DEFAULT);
                String decryptedString = new String(cipher.doFinal(cipherText), "UTF-8");
                return decryptedString;
            } catch (Exception e) {
                Log.e(TAG, "decrypt: ", e);
            }
        }
        return null;
    }
}

File: app/src/main/java/aepsapp/easypay/com/aepsandroid/common/EncryptionHelper.kt
package aepsapp.easypay.com.aepsandroid.common


import android.text.TextUtils
import android.util.Base64
import java.nio.charset.Charset

import javax.crypto.Cipher
import javax.crypto.spec.SecretKeySpec

/**
 * Created by ADMIN on 7/19/2016.
 */
object EncryptionHelper {

    private val TAG = "EncryptionHelper"


    fun encrypt(plainText: String, key: String): String {
        if (!TextUtils.isEmpty(plainText)) {
            try {
                val authKey = key.toByteArray()
                val cipher = Cipher.getInstance("AES/ECB/PKCS5Padding")
                val secretKey = SecretKeySpec(authKey, "AES")
                cipher.init(Cipher.ENCRYPT_MODE, secretKey)
                val cipherText = cipher.doFinal(plainText.toByteArray(charset("UTF8")))
                val encryptedString = String(Base64.encode(cipherText, Base64.DEFAULT), Charset.forName("UTF-8"))
                return encryptedString.replace("\n".toRegex(), "")
            } catch (e: Exception) {
                Log.e(TAG, "encrypt: ", e)
            }

        }
        return ""
    }

    fun decrypt(encryptedText: String, key: String): String {
        if (!TextUtils.isEmpty(encryptedText)) {
            try {
                val authKey = key.toByteArray()
                val cipher = Cipher.getInstance("AES/ECB/PKCS5PADDING")
                val secretKey = SecretKeySpec(authKey, "AES")
                cipher.init(Cipher.DECRYPT_MODE, secretKey)
                val cipherText = Base64.decode(encryptedText.toByteArray(charset("UTF8")), Base64.DEFAULT)
                return String(cipher.doFinal(cipherText), Charset.forName("UTF-8"))
            } catch (e: Exception) {
                Log.e(TAG, "decrypt: ", e)
            }

        }
        return ""
    }
}
